### PR TITLE
feat: step block node stories

### PR DESCRIPTION
**Pull request by levi-zustiak**
# Description
- Adds stories for `StepBlockNode`

[Link](https://3.basecamp.com/3105655/buckets/36562260/todos/7150977911) to Basecamp Todo

### CODE REVIEW

**Review by csiyang — CHANGES_REQUESTED** [review]
This story needs a few more variants
- default: should have no `image`, `title` or `description`
- One variant for each of the different types `<StepBlockNodeIcon />` (Video, TextResponse, Button, Typography, RadioQuestion)
- Hover: to show the menu fab appearing

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+  ...Template,
+  args: {
+    ...defaultFlowProps,
+    nodes: [signUpNode],
+    nodeTypes: {
+      StepBlock: StepBlockNode
+    }
+  }
+}
+
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx:833]: Can we also add the hover variant

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,834 @@
...
+const defaultFlowProps = {
+  edges: [],
+  edgeTypes: {},
+  onConnectStart: () => undefined,
+  onConnectEnd: () => undefined,
+  fitView: true,
+  proOptions: { hideAttribution: true }
+}
+
+const Template: StoryObj<ComponentPropsWithoutRef<typeof StepBlockNode>> = {
```
[the remark this answers is not in the document]
Comment (levi-zustiak) [reply apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx:754]: It was recommended in the ts doc. It was recommended to use the `ComponentWithRef` for components with a forwarded ref or `ComponentWithoutRef` for components without

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx (hunk 1 above)
Comment (levi-zustiak) [reply apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx:833]: The Menu Fab shows on hover for the other variants. Are you saying that you want a variant to display the Menu Fab without having to hover on the component?

Comment (csiyang) [reply apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx:833]: @levi-zustiak you can use story book play to mouse over the component on render, this means we can get a snapshot of the hover

**Review by csiyang — APPROVED** [review]

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,849 @@
...
+  args: {
+    ...defaultFlowProps,
+    nodes: [signUpNode],
+    nodeTypes: {
+      StepBlock: StepBlockNode
+    }
+  }
+}
+
+export const Hover = {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.stories.tsx:835]: This story is failing

**Review by csiyang — CHANGES_REQUESTED** [review]
Lint is failing

**Review by csiyang — APPROVED** [review]

**Pull request merged**